app/controllers/service_calendars_controller.rb

```diff
@@ -139,60 +106,72 @@ def move_visit_position
+
+    if params[:check] && !@locked
+      unit_minimum = @line_items_visit.line_item.service.displayed_pricing_map.unit_minimum
+
+      @visits.update_all(quantity: unit_minimum, research_billing_qty: unit_minimum, insurance_billing_qty: 0, effort_billing_qty: 0)
+    elsif params[:uncheck] && !@locked
+      @visits.update_all(quantity: 0, research_billing_qty: 0, insurance_billing_qty: 0, effort_billing_qty: 0)
     end
 
     # Update the sub service request only if we are not in dashboard; admin's actions should not affect the status
-    unless @portal
-      sub_service_request = @line_items_visit.line_item.sub_service_request
-      sub_service_request.update_attribute(:status, "draft")
+    unless @admin || @locked
+      @line_items_visit.sub_service_request.update_attribute(:status, "draft")
       @service_request.update_attribute(:status, "draft")
     end
 
-    render partial: 'update_service_calendar'
+    respond_to do |format|
+      format.js
+    end
   end
 
   def toggle_calendar_column
-    column_id  = params[:column_id].to_i
-    @arm       = Arm.find(params[:arm_id])
-    @portal    = params[:portal] == 'true'
-
-    @service_request.service_list(false).each do |_key, value|
-      next unless @sub_service_request.nil? || @sub_service_request.organization.name == value[:process_ssr_organization_name] || @sub_service_request.can_be_edited?
-
-      @arm.line_items_visits.each do |liv|
-        next if value[:line_items].exclude?(liv.line_item) || (!@portal && (!liv.line_item.sub_service_request.can_be_edited? || liv.line_item.sub_service_request.is_complete?))
-        visit = liv.visits[column_id - 1] # columns start with 1 but visits array positions start at 0
-        if params[:check]
-          visit.update_attributes quantity: liv.line_item.service.displayed_pricing_map.unit_minimum, research_billing_qty: liv.line_item.service.displayed_pricing_map.unit_minimum, insurance_billing_qty: 0, effort_billing_qty: 0
-        elsif params[:uncheck]
-          visit.update_attributes quantity: 0, research_billing_qty: 0, insurance_billing_qty: 0, effort_billing_qty: 0
-        end
+    @admin              = params[:admin] == 'true'
+    @tab                = 'template'
+    @page               = params[:page]
+    @visit_group        = VisitGroup.find(params[:visit_group_id])
+    @arm                = @visit_group.arm
+    @line_items_visits  = @arm.line_items_visits.eager_load(line_item: [:admin_rates, service: [:pricing_maps, organization: [:pricing_setups, parent: [:pricing_setups, parent: [:pricing_setups, parent: :pricing_setups]]]], service_request: :protocol])
+    @visit_groups       = @arm.visit_groups.paginate(page: @page.to_i, per_page: VisitGroup.per_page).eager_load(visits: { line_items_visit: { line_item: [:admin_rates, service: [:pricing_maps, organization: [:pricing_setups, parent: [:pricing_setups, parent: [:pricing_setups, parent: :pricing_setups]]]], service_request: :protocol] } })
+
```

Comment:
Instead of using the service_list method, only load the necessary data through the editable SubServiceRequests and update the corresponding visits in the column